prov/verbs/src/verbs_cm.c

```diff
@@ -372,10 +379,10 @@ fi_ibv_msg_xrc_ep_connect(struct fid_ep *ep, const void *addr,
 	fastlock_acquire(&xrc_ep->base_ep.eq->lock);
 	xrc_ep->conn_setup->conn_tag = VERBS_CONN_TAG_INVALID;
 	fi_ibv_eq_set_xrc_conn_tag(xrc_ep);
-	fastlock_release(&xrc_ep->base_ep.eq->lock);
-
 	dst_addr = rdma_get_peer_addr(_ep->id);
```

Comment:
> Is it required that the resolution of the peer address should be performed within the critical section covered by the lock?

If you mean rdma_get_peer_addr(), that is just an accessor that returns the value. If you mean fi_ibv_connect_xrc() thenI was trying to protect the XRC EP state and CM ID state. One issue with XRC is due to the shared nature of the XRC INI sharing, CM processing of one XRC EP can affect the state of another XRC EP through scheduling of that EP. This is trying to serialize this CM processing from the API calls that could occur that could modify XRC EP state as well. If you don't believe it is needed, I can take it out for now.

Reply:
Ok, sorry. Yes all it does just return a private member from the ID. At closer look though, the call does not appear to even be required since the called function gets the address from the ID as well.